cpp/src/util/main/UnitTests.cpp

```diff
@@ -125,6 +136,78 @@ TEST(eigen_util, sort_columns) {
   }
 }
 
+TEST(eigen_util, sort_columns_one_element) {
+  constexpr int kNumRows = 1;
+  constexpr int kMaxNumCols = 1;
+  using Array = Eigen::Array<float, kNumRows, Eigen::Dynamic, 0, kNumRows, kMaxNumCols>;
+
+  Array array{
+      {3},
+  };
+
+  array = eigen_util::sort_columns(array);
+
+  Array expected{
+      {3},
+  };
+
+  for (int r = 0; r < array.rows(); r++) {
+    for (int c = 0; c < array.cols(); c++) {
+      EXPECT_EQ(array(r, c), expected(r, c));
+    }
+  }
+}
+
+TEST(eigen_util, extract_dim_v) {
+  using Sizes = Eigen::Sizes<123, 456, 789>;
+  constexpr int N = eigen_util::extract_dim_v<1, Sizes>;
+  EXPECT_EQ(N, 456);
+
+}
+
+TEST(eigen_util, UniformDirchletGen) {
+  constexpr int M = 1e5;
+  constexpr int N = 10;
+  float alpha = 0.1;
+
+  eigen_util::UniformDirichletGen<float> gen;
+  Eigen::Rand::P8_mt19937_64 rng{ 35 };
+
+  
+  Eigen::MatrixXf X(M, N);
+  X.setZero();
+
+  for (int i = 0; i < M; ++i) {
+    X.row(i) = gen.template generate<Eigen::Array<float, N, 1>>(rng, alpha).transpose();
+  }
```

Comment:
In a unit-test, we don't really care about performance, so this is fine as is.

However, in this project, you should generally get into the habit of using fixed-size Eigen types wherever you can, so might as well develop that habit here. You should be able to replace lines 177-182 with:

```
using MatrixT = Eigen::Matrix<float, M, N>;
MatrixT X = gen.template generate<MatrixT>(rng, alpha);
```

(arguably, the template parameter typename of `Array` for the `generate()` method should be changed to reflect that an `Array` OR `Matrix` can be passed in, but `ArrayOrMatrix` is awkward, and I can't think of a good alternate name, so it's probably fine to just leave it as is.)

Reply:
Note: my suggestion here would preclude option 1 out of the 3 options I listed for `computeCovariance()`.